### PR TITLE
Added rank-properties to RankProfile and RankProperty class

### PR DESCRIPTION
**Pull request by JeanExtreme002**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: vespa/package.py [hunk 1]
```diff
@@ -1432,6 +1473,7 @@ def __init__(
         self.model_configs = {}
         self.stateless_model_evaluation = stateless_model_evaluation
         self.models = {}
+        self.user_version=user_version
```
Comment (lesters) [inline vespa/package.py:1476]: PR test fail here, user_version is no longer an argument.

**Review by lesters — CHANGES_REQUESTED** [review]
PR test fails on remnant of user_version

**Comment by JeanExtreme002** [comment]
Done! @lesters

**Comment by lesters** [comment]
Thanks, great. Now, there are som merge conflicts, due to other recently merged PRs, Could you take a look?

**Comment by JeanExtreme002** [comment]
@lesters I'm closing this PR because [maxice8](https://github.com/maxice8) already implemented rank-properties and other features at his PR.

**Pull request closed without merging**